build-vm-qemu

```diff
@@ -224,13 +223,14 @@ vm_fixup_qemu() {
             ;;
     esac
 
-    if test -z "$qemu_serial_device" ; then
-        if test -e "$BUILD_ROOT/.build.console.kvm" -a ! -L "$BUILD_ROOT/.build.console.kvm" && grep -q '^virtio$' "$BUILD_ROOT/.build.console.kvm" ; then
-            echo "Detected virtio-serial support"
-            qemu_serial_device=virtio-serial,max_ports=2
-            qemu_console=hvc0
-        fi
-    fi
+# virtio serial built-in seems not to work since qemu 5.2 anymore
```

Comment:
it works just fine here? I think the problem you're facing is that `vm_startup_qemu()` is overriding `qemu_console` with ttyAMA0 rather than keeping `qemu_console` at hvc0 like set here.

the underlying issue is that the bulk of the qemu* device determination logic needs to be executed much earlier, like in the kvm case it is done in `vm_verify_options_kvm` which however is too early for qmeu as the `$BUILD_HOST_ARCH` isn't set at that point. 

I see two options: move determination of BUILD_HOST_ARCH much earlier and move the device determination logic into `vm_verify_options_qemu`. the other option I see is that we share move about the qemu_storage device and drive as well as the virtio serial determination logic with the kvm case. 

by a quick hack I did a sed /qemu_device/kvm_device/ on build-vm-kvm and that makes it work for me. obviously thats not a nice fix but the virtio detection and serial detection logic all works naturally then.

Reply:
it was indeed the kernel parameter. I changed the code now to avoid overwriting an already set console name.